schema/changelog-3.6.xml

```diff
@@ -29,6 +30,65 @@
     <addColumn tableName="devices">
       <column name="motion" type="VARCHAR(128)" />
     </addColumn>
+    <addColumn tableName="devices">
+      <column name="geofenceId" type="INT" />
+    </addColumn>
+
+    <createTable tableName="geofences">
+      <column name="id" type="INT" autoIncrement="true">
+        <constraints primaryKey="true" />
+      </column>
+      <column name="name" type="VARCHAR(128)">
+        <constraints nullable="false" />
+      </column>
+      <column name="description" type="VARCHAR(128)" />
+      <column name="area" type="VARCHAR(4096)">
+        <constraints nullable="false" />
+      </column>
+      <column name="attributes" type="VARCHAR(4096)">
+        <constraints nullable="false" />
+      </column>
+    </createTable>
+
+    <createTable tableName="user_geofence">
+      <column name="userid" type="INT">
+        <constraints nullable="false" />
+      </column>
+      <column name="geofenceid" type="INT">
+        <constraints nullable="false" />
+      </column>
+    </createTable>
+
+    <addForeignKeyConstraint baseTableName="user_geofence" baseColumnNames="userid" constraintName="fk_user_geofence_userid" referencedTableName="users" referencedColumnNames="id" onDelete="CASCADE" />
+    <addForeignKeyConstraint baseTableName="user_geofence" baseColumnNames="geofenceid" constraintName="fk_user_geofence_geofenceid" referencedTableName="geofences" referencedColumnNames="id" onDelete="CASCADE" />
+
+    <createTable tableName="group_geofence">
+      <column name="groupid" type="INT">
+        <constraints nullable="false" />
+      </column>
+      <column name="geofenceid" type="INT">
+        <constraints nullable="false" />
+      </column>
+    </createTable>
+
+    <addForeignKeyConstraint baseTableName="group_geofence" baseColumnNames="groupid" constraintName="fk_group_geofence_groupid" referencedTableName="groups" referencedColumnNames="id" onDelete="CASCADE" />
+    <addForeignKeyConstraint baseTableName="group_geofence" baseColumnNames="geofenceid" constraintName="fk_group_geofence_geofenceid" referencedTableName="geofences" referencedColumnNames="id" onDelete="CASCADE" />
+
+    <createTable tableName="user_device_geofence">
+      <column name="userid" type="INT">
```

Comment:
I'll try to explain.
user_geofences - it is permissions - user can edit, delete and connect this geofences

user_device_geofence - it is link device-geofence, but per user. It solves the case then two users access to the same geofence but need to know enter/exit about two different devices.

For example: a sales company, a stockman need to be notified when comes semi-truck to warehouse to be prepared for unload it, sales manager need to be notified when other van leaved warehouse for delivery to call client. They both have access to "warehouse" geofence and both have access to semi-truck and van, but only one want to be notified about truck and other about van.

It just allow users to be flexible notified about only some of their devices entered/exited geofences.